src/Microsoft.AspNetCore.Razor.LanguageServer.Common/ProjectEngineFactory_3_0.cs

```diff
@@ -2,28 +2,35 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Linq;
 using System.Reflection;
 using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
 using Microsoft.CodeAnalysis.Razor;
 
-namespace Microsoft.AspNetCore.Razor.LanguageServer
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common
 {
     internal class ProjectEngineFactory_3_0 : IProjectEngineFactory
     {
         private const string AssemblyName = "Microsoft.AspNetCore.Mvc.Razor.Extensions";
+
         public RazorProjectEngine Create(RazorConfiguration configuration, RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure)
         {
             // Rewrite the assembly name into a full name just like this one, but with the name of the MVC design time assembly.
-            var assemblyName = new AssemblyName(typeof(ProjectEngineFactory_3_0).Assembly.FullName);
+            var assemblyName = new AssemblyName(typeof(RazorProjectEngine).Assembly.FullName);
             assemblyName.Name = AssemblyName;
 
             var extension = new AssemblyExtension(configuration.ConfigurationName, Assembly.Load(assemblyName));
             var initializer = extension.CreateInitializer();
 
             return RazorProjectEngine.Create(configuration, fileSystem, b =>
             {
+                CompilerFeatures.Register(b);
+
                 initializer.Initialize(b);
                 configure?.Invoke(b);
+
+                b.Features.OfType<ComponentDocumentClassifierPass>().Single().MangleClassNames = true;
```

Comment:
This was the line that throws for this [bug](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/788091). Did you run into this at any point?

Reply:
This line has thrown a few times. That being said I've been repairing the airplane while it's flying so when I encountered the issue I was in the midst of fixing other issues and didn't directly correlate the exception (I assumed it was my other work). The exception stopped and I haven't ran into it for a while.